### DISCUSSION TITLE
Auto-unpublish new katas with issues

### DISCUSSION THREAD

**Discussion by Kacarott** [codewars/codewars.com#2600] [Ideas]
Some new beta katas when published, have big problems. I would loosely classify such katas into two groups: legitimate authors trying to make a quality kata, but who are inexperienced or have overlooked something; and "low effort" authors who put in almost no time, perhaps in an attempt to collect solutions for homework, or similar. 

I propose, that a functionality similar to auto-retire be added (ie. auto-unpublish). Any "new" kata (the threshold for this is arbitrary, perhaps less than 48 hours old?) which collects some number of issues (2?) would be automatically sent back to draft, where [this suggestion](https://github.com/codewars/codewars.com/discussions/2598) would ensure that the author works on fixing the issues.

The idea provides a win-win scenario. 

- Authors who are genuinely trying to make a good kata, would be spared the initial volley of unsatisfied votes due to an overlooked issue, giving them time to recognize the errors and fix them. It is already common for mods to revert betas for this reason, so an auto-unpublish feature would mean less work for them, and more consistency for authors.
- "Low effort" katas could also be removed from beta quickly and easily, without requiring 4 users each to solve and vote unsatisfied (as is the current state). Authors of these katas would be forced to put effort in, otherwise leaving their kata in draft mode with no solves. This also solves #2560 because katas could be easily removed from beta, by a single user (with good reason -> issues), without providing even a single solution.

So overall, the system would protect inexperienced, but "genuinely trying" authors, and also provide much better means to deal with low effort/homework katas.

**Comment by JohanWiltink** [comment]
Two issues _seems_ few, but no irreversible damage is done, so that is probably fine.

**Comment by kazk** [comment]
Sounds good to me. We can discuss the details later, like the number of issues to trigger this.

**Comment by kazk** [comment]
Recently published kata (within the past 48h) is now automatically unpublished if it has 2 unresolved issues. We can adjust the numbers after trying this out.
